homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,356 @@
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import GroupEntity
+from .util import (
+    find_state_attributes,
+    most_frequent_attribute,
+    reduce_attribute,
+    states_equal,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Climate Group"
+
+# No limit on parallel updates to enable a group calling another group
+PARALLEL_UPDATES = 0
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_TEMPERATURE_UNIT, default=TEMP_CELSIUS): cv.string,
+        vol.Required(CONF_ENTITIES): cv.entities_domain(DOMAIN),
+    }
+)
+# edit the supported_flags
+SUPPORT_FLAGS = (
+    ClimateEntityFeature.TARGET_TEMPERATURE
+    | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+    | ClimateEntityFeature.PRESET_MODE
+    | ClimateEntityFeature.SWING_MODE
+    | ClimateEntityFeature.FAN_MODE
+)
+
+
+# HVAC Action priority
+HVAC_ACTIONS_PRIORITY = [
+    HVACAction.HEATING,
+    HVACAction.COOLING,
+    HVACAction.DRYING,
+    HVACAction.FAN,
+    HVACAction.IDLE,
+    HVACAction.OFF,
+    None,
+]
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Initialize climate.group platform."""
+    async_add_entities(
+        [
+            ClimateGroup(
+                config.get(CONF_UNIQUE_ID),
+                config[CONF_NAME],
+                config[CONF_ENTITIES],
+                config[CONF_TEMPERATURE_UNIT],
+            )
+        ]
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Light Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+
+    async_add_entities(
+        [
+            ClimateGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options[CONF_TEMPERATURE_UNIT],
+            )
+        ]
+    )
+
+
+class ClimateGroup(GroupEntity, ClimateEntity):
+    """Representation of a climate group."""
+
+    _attr_available: bool = False
+    _attr_assumed_state: bool = True
+
+    def __init__(
+        self, unique_id: str | None, name: str, entity_ids: list[str], unit: str
+    ) -> None:
+        """Initialize a climate group."""
+        self._entity_ids = entity_ids
+
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+
+        if "c" in unit.lower():
```

Comment:
Can we pass `TEMP_CELSIUS` and `TEMP_FAHRENHEIT` in directly? That way we only have a single-line assignment